containers/sip-servlets-as7/src/main/java/org/jboss/as/clustering/web/infinispan/sip/DistributedCacheConvergedSipManagerFactory.java

```diff
@@ -0,0 +1,341 @@
+import org.jboss.as.clustering.infinispan.subsystem.AbstractCacheConfigurationService;
+import org.jboss.as.clustering.infinispan.subsystem.CacheService;
+import org.jboss.as.clustering.infinispan.subsystem.EmbeddedCacheManagerService;
+import org.jboss.as.clustering.lock.SharedLocalYieldingClusterLockManager;
+import org.jboss.as.clustering.lock.impl.SharedLocalYieldingClusterLockManagerService;
+import org.jboss.as.clustering.msc.AsynchronousService;
+import org.jboss.as.clustering.registry.Registry;
+import org.jboss.as.clustering.registry.RegistryService;
+import org.jboss.as.clustering.web.BatchingManager;
+import org.jboss.as.clustering.web.ClusteringNotSupportedException;
+import org.jboss.as.clustering.web.LocalDistributableSessionManager;
+import org.jboss.as.clustering.web.OutgoingDistributableSessionData;
+import org.jboss.as.clustering.web.SessionAttributeMarshaller;
+import org.jboss.as.clustering.web.SessionAttributeMarshallerFactory;
+import org.jboss.as.clustering.web.impl.SessionAttributeMarshallerFactoryImpl;
+import org.jboss.as.clustering.web.impl.TransactionBatchingManager;
+import org.jboss.as.clustering.web.infinispan.SessionAttributeStorage;
+import org.jboss.as.clustering.web.infinispan.SessionAttributeStorageFactory;
+import org.jboss.as.clustering.web.infinispan.SessionAttributeStorageFactoryImpl;
+import org.jboss.as.clustering.web.infinispan.WebSessionCacheConfigurationService;
+import org.jboss.as.clustering.web.sip.DistributedConvergedCacheManagerFactoryService;
+import org.jboss.logging.Logger;
+import org.jboss.metadata.web.jboss.JBossWebMetaData;
+import org.jboss.metadata.web.jboss.ReplicationConfig;
+import org.jboss.msc.service.Service;
+import org.jboss.msc.service.ServiceBuilder;
+import org.jboss.msc.service.ServiceBuilder.DependencyType;
+import org.jboss.msc.service.ServiceController;
+import org.jboss.msc.service.ServiceName;
+import org.jboss.msc.service.ServiceRegistry;
+import org.jboss.msc.service.ServiceTarget;
+import org.jboss.msc.value.InjectedValue;
+import org.jboss.tm.XAResourceRecoveryRegistry;
+import org.jboss.as.clustering.web.infinispan.DistributedCacheManagerFactory;
+
+/**
+ * 
+ * @author posfai.gergely@ext.alerant.hu
+ * 
+ */
+public class DistributedCacheConvergedSipManagerFactory extends DistributedCacheManagerFactory {
+	
+	private static final Logger logger = Logger.getLogger(DistributedCacheConvergedSipManagerFactory.class);
+	
+	//public static final String DEFAULT_CACHE_CONTAINER = "sip";
+
+	private static final ServiceName JVM_ROUTE_REGISTRY_SERVICE_NAME = DistributedConvergedCacheManagerFactoryService.JVM_ROUTE_REGISTRY_ENTRY_PROVIDER_SERVICE_NAME.getParent();
+
+	private SessionAttributeStorageFactory storageFactory = new SessionAttributeStorageFactoryImpl();//TODO: do we need here a ConvergedSessionAttributeMarshallerFactory? Probably not.
+	private CacheInvoker invoker = new RetryingCacheInvoker(new BatchCacheInvoker(), 10, 100);
```

Comment:
10,100 magic numbers. Define cosntant, do we need these params to be configurable?